sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -38,6 +38,9 @@ public class ChatClient {
     private let service: Chat
     private var signalingClient: CommunicationSignalingClient?
     private var signalingClientStarted: Bool = false
+    private var pushNotificationClient: PushNotificationClient
+    private var pushNotificationsStarted: Bool = false
+    internal var registrationId: String
```

Comment:
From what I can remember, we decided to put "registrationId" in ChatClient to pass the ChatClientDVR test.

On 2/25/2022's chat discussion, you suggests that - "You could actually have ChatClient generate the GUID when you create the client, and then when you call startNotifications it uses that and generates a new one. It has the same effect but then you could change the value to something deterministic in your test and register the replacement."

Reply:
And for "pushNotificationsStarted", we need this external to do some "pre-validation". For example, when Contoso calls chatClient.startPushNotification() and finds the PN has already started, Contoso does not need to initialize the PushNotificationClient at all.

Reply:
I think my advice from 2/25 still applies if the registrationId lives in the pushNotificationClient. For the second part, it would still work if `started` was part of the notification client.
```swift
func startPushNotification(...) {
  guard self.pushNotificationClient?.started == false else {
    "Push notifications already started".
  }
}
```

etc.

Reply:
Yes you are right. I put both of them inside PN client then.